megamek/src/megamek/client/ui/swing/tooltip/UnitToolTip.java

```diff
@@ -179,62 +185,131 @@ private static boolean hideArmorLocation(Entity entity, int location) {
     }
 
     private static String locationHeader(Entity entity, int location) {
-        return entity.isConventionalInfantry() ?
-                ((Infantry) entity).getShootingStrength() + " Active Troopers" : entity.getLocationAbbr(location);
+        String msg_activetroopers =Messages.getString("BoardView1.Tooltip.ActiveTroopers");
+        return entity.isConventionalInfantry() ? ((Infantry) entity).getShootingStrength() + " " + msg_activetroopers : entity.getLocationAbbr(location);
+    }
+
+    private static StringBuilder sysCrits(Entity entity, int type, int index, int loc, String l) {
+        StringBuilder result = new StringBuilder();;
+        int good = entity.getGoodCriticals(type, index, loc);

```

Comment:
This adds e.g. 6 crits for the engine. Would it make sense to only show the crits necessary to destroy the unit instead (as on the record sheet)? This would probably require some rewriting so I'm okay with it as it is now but it crossed my mind when seeing the engine crits in the screenshots.

Reply:
think xl engines with slots in the left and right torso would also make this complicated